library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Record.hs

```diff
@@ -1 +1,132 @@
-module Juvix.Backends.LLVM.Codegen.Record where
+module Juvix.Backends.LLVM.Codegen.Record
+  ( register,
+    loadField,
+    makeRecord,
+  )
+where
+
+import qualified Data.List as List
+import qualified Juvix.Backends.LLVM.Codegen.Block as Block
+import qualified Juvix.Backends.LLVM.Codegen.Types as Types
+import qualified Juvix.Backends.LLVM.Pass.Types as PassTypes
+import Juvix.Library hiding (Type, local)
+import qualified Juvix.Library.HashMap as Map
+import Juvix.Library.NameSymbol as NameSymbol
+import qualified LLVM.AST as AST
+import qualified LLVM.AST.Type as Type
+import qualified Prelude as P
+
+-- | @register@ registers a record type with the given name and
+-- | field names and types
+register ::
+  Types.Define m =>
+  PassTypes.RecordName ->
+  [PassTypes.FieldName] ->
+  [Type.Type] ->
+  m ()
+register recordName fieldNames llvmFieldTypes = do
+  let llvmTypeName = recordTypeName recordName
+  Block.addType llvmTypeName (llvmRecordType llvmFieldTypes)
+  recordTable <- get @"recordTab"
+  let typeRef = Type.NamedTypeReference llvmTypeName
+  let fieldDescs = zip (map toSymbol fieldNames) llvmFieldTypes
+  put @"recordTab" $ Map.insert (toSymbol recordName) (typeRef, fieldDescs) recordTable
```

Comment:
Why not
```haskell
  let typeRef = Type.NamedTypeReference llvmTypeName
  let fieldDescs = zip (map toSymbol fieldNames) llvmFieldTypes
  modify @"recordTab" (Map.insert (toSymbol recordName) (typeRef, fieldDescs))

Reply:
Good call, done in https://github.com/anoma/juvix/pull/1106/commits/fe9d23d65740d4570a9d00a47aae9545cb4d3475